nyxx.interactions/lib/src/models/Interaction.dart

```diff
@@ -0,0 +1,50 @@
+part of nyxx_interactions;
+
+class Interaction extends SnowflakeEntity implements Disposable {
+  /// Reference to bot instance
+  final Nyxx client;
+
+  late final InteractionType type;
+
+  late final CommandInteractionData? data;
+
+  late final Snowflake guild_id;
+
+  late final Snowflake channel_id;
+
+  late final User author;
+
+  late final String token;
+
+  late final int version;
+
+  Interaction._new(
+    this.client,
+    Map<String, dynamic> raw,
+  ) : super(Snowflake(raw["id"])) {
+    switch (raw["type"] as int) {
+      case 1:
+        {
```

Comment:
I dont like these braces as they are not needed here

Reply:
No longer required due to removed use of Enums, removed in next push